pkg/util/kubernetes/clustername/clustername.go

```diff
@@ -120,6 +126,17 @@ func GetClusterName(ctx context.Context, hostname string) string {
 	return getClusterName(ctx, defaultClusterNameData, hostname)
 }
 
+// GetRFC1123CompliantClusterName returns a k8s cluster name if it exists, either directly specified or autodiscovered
+// Some kubernetes cluster-names (EKS,AKS) are not RFC1123 compliant, mostly due to an `_`.
+// This function replaces the invalid `_` with a valid `-`.
+func GetRFC1123CompliantClusterName(ctx context.Context, hostname string) string {
+	clusterName := getClusterName(ctx, defaultClusterNameData, hostname)
+	if strings.Contains(clusterName, "_") {
+		log.Warnf("clustername: %s contains `_`, replacing it with `-` to be hostName compliant", clusterName)
```

Comment:
I think this message is a bit misleading. We're doing replacement for host alias, but this warning mentions "cluster name" and "host name", neither of which are affected by replacement.

Since this logic is required only for host alias, would it make sense to move this into `hostname.go` and make it private? Then the warning could be something like: `"hostAlias: cluster name '%s' contains '_', replacing it with '-' to be RFC1123 compliant"`. What do you think?

Reply:
hm, yes indeed that sounds better. Let me update it. 👍

Reply:
Added here: https://github.com/DataDog/datadog-agent/pull/9279/commits/1b0aa72ec5dbd86bede4f2624649247b29cc4874